lib/wolfe/cleanup.rb

```diff
@@ -50,31 +52,46 @@ def timespan_regex
       #
 
       def cleanup( config )
-        daily_date = Date.today - eval( config['one_per_day_timespan'] )
-        monthly_date = Date.today - eval( config['one_per_month_timespan'] )
-        first_relevant_date = Date.today - 5.years
+        daily_date = Date.today - eval(config['one_per_day_timespan'])
+        monthly_date = calculate_monthly_date_for_backup_to_keep(config['one_per_month_timespan'])
+        keep_one = !one_per_month_timespan_starts_with_zero?(config['one_per_month_timespan'])
 
         if File.directory?(config['path'])
-          clean_monthly( monthly_date, daily_date, config )
-          clean_yearly( first_relevant_date, monthly_date, config )
+          clean_monthly(monthly_date, daily_date, config, keep_one)
+          clean_yearly(monthly_date, config, keep_one) if keep_one
         else
           puts "Path '#{config['path']}' is not a directory."
         end
       end
 
-      def clean_monthly( monthly_date, daily_date, config )
-        monthly_date.upto( daily_date ) do |date|
-          delete_but_keep_one_per_month( config['path'], config['filename'], date )
+      def calculate_monthly_date_for_backup_to_keep(one_per_month_timespan)
+        timespan_digit = one_per_month_timespan.split('.')[0].to_i
+        timespan_word = one_per_month_timespan.split('.')[1].to_sym
+
+        if Date.today - timespan_digit.send(timespan_word) == Date.today
+          BACKUP_FIRST_CUTOFF_DATE
+        else
+          Date.today - timespan_digit.send(timespan_word)
+        end
+      end
+
+      def one_per_month_timespan_starts_with_zero?(timespan)
```

Comment:
If you go with the `TimespanFromConfig` approach above, this method could be moved into that class as well. And then rename it to something more descriptive, because I have no idea why a leading 0 is important in relation to the backups :)